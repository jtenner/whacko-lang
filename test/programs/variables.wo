export fn _start(): void {
  let a = 10;
  a = 10;
}