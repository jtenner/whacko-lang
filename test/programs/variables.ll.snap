; ModuleID = 'whacko'
source_filename = "whacko"
target triple = "wasm32-wasi"

declare void @__whacko_gc_initialize()

declare ptr @__whacko_gc_alloc(i32, i32)

declare void @__whacko_gc_store_field_ptr(i32, ptr, ptr, ptr)

define void @_start() {
entry:
  call void @__whacko_gc_initialize()
  %"alloca~0" = alloca i64, align 8
  %"64~inst" = load i64, ptr %"alloca~0", align 4
  store i64 10, ptr %"alloca~0", align 4
  ret void
}

declare void @__whacko_gc_visit(ptr)

define void @__visit_whacko_object(ptr %0) {
"entry~69":
  %"typeGEP~71" = getelementptr { ptr, ptr, i32, i32 }, ptr %0, i32 2
  %1 = load i32, ptr %"typeGEP~71", align 4
  switch i32 %1, label %"unreachable~70" [
    i32 1, label %"classType:[std/String.wo]String<>"
  ]

"unreachable~70":                                 ; preds = %"entry~69"
  unreachable

"classType:[std/String.wo]String<>":              ; preds = %"entry~69"
  ret void
}
